### PR TITLE
OFFLINE resource serving for desktop

### PR DESCRIPTION
**Pull request by artpi**
# Why?

We want to serve fonts locally because, lets face it, it makes sense.

CC @mtias

### CODE REVIEW

File: public/fonts/fonts-desktop.css [hunk 1]
```diff
@@ -0,0 +1,72 @@
+@font-face {
+  font-family: 'Merriweather';
+  font-style: normal;
+  font-weight: 400;
+  src: local('Merriweather'), url(http://localhost:41050/calypso/fonts/Merriweather.ttf) format('truetype');
```
Comment (artpi) [inline public/fonts/fonts-desktop.css]: @mtias, you think that best approach here would be webpack?

Comment (mtias) [inline public/fonts/fonts-desktop.css]: Not sure what you mean here? Once we move loading of CSS assets to webpack loaders we'll likely want all this under a common.css file.

File: server/boot/index.js [hunk 2]
```diff
@@ -45,7 +45,11 @@ function setup() {
 	}
 
 	// attach the static file server to serve the `public` dir
-	app.use( '/calypso', express.static( path.resolve( __dirname, '..', '..', 'public' ) ) );
+	app.use( '/calypso', express.static( path.resolve( __dirname, '..', '..', 'public' ), {
+		setHeaders: function( res, path ) {
```
Comment (mtias) [inline server/boot/index.js]: Were you planning on doing this only for desktop?

**Comment by mtias** [comment]
I think this makes sense, but what are the scenarios we need to cover?

* Web: should load most everything from WP.com CDN.
* Development: local would be fine, though it introduces a vector of potential divergence.
* Desktop: local all the time for offline + speed?

File: public/fonts/fonts-desktop.css (hunk 1 above)
Comment (artpi) [inline public/fonts/fonts-desktop.css]: I was thinking about WebPack supplying proper URL according to what is currently built.
But if webpack will be bundling it, then no problem.

File: server/boot/index.js (hunk 2 above)
Comment (artpi) [inline server/boot/index.js]: Initially yes, because desktop is the only version you can load fully offline

**Comment by artpi** [comment]
@mtias: How about all versions load from CDN with fallback to local if ?
We can accomplish this by adding onError listener on the `LINK`

File: public/fonts/fonts-desktop.css (hunk 1 above)
Comment (mtias) [inline public/fonts/fonts-desktop.css]: Right, Yes, I think we should move towards webpack handling this. It's about time we start looking at chunking our CSS as well.

**Comment by mtias** [comment]
@artpi that sounds like a good plan to me.

**Comment by artpi** [comment]
Whole offline plan is going on ice, so I'm closing this PR

**Pull request closed without merging**